### DISCUSSION TITLE
Understanding the Usage Summary on the Dashboard

### DISCUSSION THREAD

**Discussion by PLyn** [supabase/supabase#27479] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/understanding-the-usage-summary-on-the-dashboard-D7Gnle)._

## Organization usage metrics

This discussion explains your organization's usage metrics, which can be found here: <https://supabase.com/dashboard/org/_/usage>.

![](/docs/img/troubleshooting/46f59019-6ff1-4bfb-9acd-d5b7bd282f89.png)

These metrics are calculated based on all the projects under the organization that meet these criteria:

*   Currently active projects
*   Inactive projects that were active for some period of time within the current billing cycle

This may be confusing for users looking at only what metrics are being used for currently active projects.

## Database size example

Using the database size as an example, suppose we have the following projects under the organization:

*   Project A: 1 GB database size
*   Project B: 0.5 GB database size
*   Project C: 0.5 GB database size

If the user decides to delete Projects A and B during the current billing cycle, the average database usage will be 2 GB for the current billing cycle, even though only a 0.5 GB database is currently active.

As of now, even if your project was only active for a few days, we take the average database size for those active days, which may lead to high usage when creating and deleting many projects within a billing cycle. We're working on a better billing model to cover these cases.

## Edge Functions example

Consider another example with edge functions. Suppose we have these projects under the organization:

*   Project D: 5 Edge Functions
*   Project E: 7 Edge Functions

If Project E with its 7 functions is deleted during the current billing cycle, the average Edge Function count for this billing cycle on the dashboard will still be 12 functions.

## Other metrics

It is a little more tricky to understand metrics like realtime and egress due to being more complex to track usage compared to Database size or Function count. However, they will account for usage from all projects that are active or were active during your current billing cycle as well.

## Metrics still look weird or incorrect?

If the numbers from inactive projects carry over to the next billing cycle and no new usage has occurred, something might be wrong.

If this is the case, or if you are unsure why any usage appears the way it does for any reason, you should reach out to Support on the dashboard here: <https://supabase.com/dashboard/support/new>.

**Comment by GaryAustin1** [comment]
2GB does not equal 2.5 GB... 
![image](https://github.com/supabase/supabase/assets/54564956/c73d2220-7539-4376-8e0d-da77f7ff1c04)

The database thing seems odd.  If I delete my projects first day of billing period then as days go on the average size they used should decrease?  
Are you saying billing is based on the MAX size at any point during the billing period?  How is average of 1.5G carried on after it has been deleted?  If I delete all my data before I delete the project does that make a difference?

Your edge function makes more sense as the billing page says it is MAX edge functions during the period.   Database size says Average size during period.

**Reply by PLyn** [reply]
Thanks! Corrected. Had different figures before and forgot to correct the total amount.

Regarding database size, it is not very clear especially around the wording of average for the period. I guess a more accurate term would be total size for that period? I will see if I can raise this or correct any mentions to make it clearer that its not literally the average database size used over the period.

> Are you saying billing is based on the MAX size at any point during the billing period?

To answer this, yes, It is essentially the total database size used over the current billing period. I don't know the original intent behind this but I believe this is because you are allocated those resources over that period of time.

> If I delete all my data before I delete the project does that make a difference?

I don't believe it does because the project was allocated the full disk size and is being charged based on that and not necessarily based on how much of the space you used at that point.

The current model of billing is a little rough for those who created and used projects for smaller portions of time in the billing cycle. Those cases where multiple projects are paused or deleted are the ones who will see discrepancies that might surprise them in the dashboard due to how this works.

**Reply by kevcodez** [reply]
The current usage billing model for database size lacks in at least two cases
- Launching and deleting projects in a short time span
- On plan up/downgrades with partial billing cycles (i.e. upgrading Pro plan to Teams after 3 days of being on Pro)

We're aware of this and there will be a bigger change around billing for database size and storage size in the upcoming weeks (prorated down to the hour to cover both cases). We're also going to remove usage-billing for function counts. We'll do a bigger round of comms for these upcoming changes soon.
